File: backend/src/main/java/com/gdje_izlazimo/project/dto/request/update/UpdateVenueRequest.java
package com.gdje_izlazimo.project.dto.request.update;

import com.gdje_izlazimo.project.enums.VenueCategory;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;

public record UpdateVenueRequest(

        @NotBlank(message = "Name is required")
        @Size(min = 4, message = "Name must have minimum 4 characters")
        String name,

        String description,

        @NotNull(message = "Activity is required")
        Boolean isActive,

        @NotNull(message = "Category is required")
        VenueCategory venueType,

        @NotBlank(message = "Phone is required")
        String phone
) {
}

File: frontend/src/app/core/models/venues/update-venue.request.ts
import { VenueCategory } from "./venue-category.enum";
import { VenueKind } from "./venue-response.dto";


export interface UpdateVenueRequest{
    name: string;
    description?: string;
    isActive: boolean;
    venueType: VenueCategory;
    venueKind: VenueKind,
    phone: string;
    instagram?: string,
    addressName: string;

}